inc/namespace.php

```diff
@@ -262,3 +262,91 @@ function autoenqueue( string $manifest_path, string $target_bundle, array $optio
 		wp_enqueue_style( $handle );
 	}
 }
+
+/**
+ * Helper function to naively check whether or not a given URI is a CSS resource.
+ *
+ * @param string $uri A URI to test for CSS-ness.
+ * @return boolean Whether that URI points to a CSS file.
+ */
+function is_css( string $uri ) : bool {
+	return preg_match( '/\.css(\?.*)?$/', $uri ) === 1;
+}
+
+/**
+ * Attempt to register a particular script bundle from a manifest.
+ *
+ * @param string $manifest_path File system path for an asset manifest JSON file.
+ * @param string $target_asset  Asset to retrieve within the specified manifest.
+ * @param array  $options {
+ *     @type string $handle       Handle to use when enqueuing the asset. Required.
+ *     @type array  $dependencies Script or Style dependencies. Optional.
+ * }
+ */
+function register_asset( string $manifest_path, string $target_asset, array $options = [] ) : void {
+	$defaults = [
+		'dependencies' => [],
+	];
+	$options = wp_parse_args( $options, $defaults );
+
+	$manifest_folder = trailingslashit( dirname( $manifest_path ) );
+
+	$asset_uri = Manifest\get_manifest_resource( $manifest_path, $target_asset );
+
+	// If we fail to match a .css asset, try again with .js in case there is a
+	// JS wrapper for that asset available (e.g. when using DevServer).
+	if ( empty( $asset_uri ) && is_css( $target_asset ) ) {
+		$asset_uri = Manifest\get_manifest_resource( $manifest_path, preg_replace( '/\.css$/', '.js', $target_asset ) );
```

Comment:
This is the piece of this that I'm least confidant about: it solves #1 because you can say
```register_asset( $manifest_path, 'some-file.css', [ 'handle' => 'my-css' ] );```
but it's potentially unintuitive that this could lead to the enqueuing of a JS file in some circumstances.

There's also a small but non-zero chance that, in the event that you try to enqueue a CSS file while running the dev server, and you wish to enqueue the CSS in a different context or condition than the JS file from which the CSS will be extracted, that you could end up having the JS run in an unexpected situation if you try to enqueue the CSS without the JS (impossible when they haven't been split apart using MiniCSSExtractPlugin). I cannot think off hand of a time when this would practically happen on a project, but as much as moving away from `autoenqueue` distances us from the :sparkles: magic :sparkles: of this library, the inference of a JS file when you asked for CSS still is "magic" and may be confusing.

Or, am I over-thinking this :grin:

Reply:
This has tricky implications for dependency handling. We may need to revert to the `scripts` and `styles` arrays for dependencies rather than tracking it as one bundle, or else reinstate some of the `$has_css` logic from the prior versions of this function.